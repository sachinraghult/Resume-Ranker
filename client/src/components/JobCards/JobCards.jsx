import React, { useState, useEffect, useContext } from "react";
import { useNavigate } from "react-router-dom";
import axios from "axios";
import { SERVER_URL } from "../../config";
import { Context } from "../../context/Context";
import styles from "./JobCards.module.css";

function JobCards(props) {
  let navigate = useNavigate();

  const { authToken } = useContext(Context);

  const [application, setApplication] = useState();

  const monthNames = [
    "January",
    "February",
    "March",
    "April",
    "May",
    "June",
    "July",
    "August",
    "September",
    "October",
    "November",
    "December",
  ];

  const dayNames = [
    "Sunday",
    "Monday",
    "Tuesday",
    "Wednesday",
    "Thursday",
    "Friday",
    "Saturday",
  ];

  const imageCollection = [
    "/Hiring/one.jpg",
    "/Hiring/two.jpg",
    "/Hiring/three.jpg",
    "/Hiring/four.png",
    "/Hiring/five.jpg",
    "/Hiring/six.jpg",
    "/Hiring/seven.jpg",
    "/Hiring/eight.jpg",
    "/Hiring/nine.jpg",
    "/Hiring/ten.jpg",
  ];

  function randomIntFromInterval(min, max) {
    return Math.floor(Math.random() * (max - min + 1) + min);
  }

  const handleApply = (e) => {
    navigate("/apply/" + props.post._id);
  };

  function onEmailClick() {
    window.open(`mailto:${props.post.email}`);
  }

  useEffect(() => {
    const fetchApplication = async () => {
      try {
        const res = await axios.get(
          `${SERVER_URL}/application/post/` + props.post._id,
          {
            headers: { Authorization: authToken },
          }
        );
        console.log(res.data);
        setApplication(res.data);
      } catch (err) {}
    };
    fetchApplication();
  }, []);

  return (
    <div className={styles.jobcard}>
      <div className={styles.jobcardheader}>
        <div>
          <img
            className={styles.cardImg}
            src={
              window.location.origin +
              imageCollection[randomIntFromInterval(0, 9)]
            }
            alt="job poster"
          />
        </div>
        <div className={styles.deadline}>
          <div style={{ fontSize: "14px" }}>
            <b>Deadline</b>
          </div>
          <div>
            <time
              className={`${styles.dateAsCalendar} ${styles.positionPixels}`}
            >
              <span className={styles.weekday}>
                {props.post
                  ? dayNames[new Date(props.post.deadline).getDay()]
                  : null}
              </span>
              <span className={styles.day}>
                {props.post ? new Date(props.post.deadline).getDate() : null}
              </span>
              <span className={styles.month}>
                {props.post
                  ? monthNames[new Date(props.post.deadline).getMonth()]
                  : null}
              </span>
              <span className={styles.year}>
                {props.post
                  ? new Date(props.post.deadline).getFullYear()
                  : null}
              </span>
            </time>
          </div>
        </div>
      </div>

      <div className={styles.jobdetailbuttons}>
        {props?.post?.applied ? (
          <button
            className={`${styles.searchbuttons} ${styles.detailbutton}`}
            style={{
              backgroundColor:
                application?.status === "In Review"
                  ? "rgb(245, 214, 157, 0.4)"
                  : application?.status === "Accepted"
                  ? "rgb(184, 236, 184, 0.4)"
                  : "rgb(244, 165, 165, 0.4)",
              color:
                application?.status === "In Review"
                  ? "orange"
                  : application?.status === "Accepted"
                  ? "green"
                  : "red",
            }}
          >
            {application?.status}
          </button>
        ) : null}
      </div>

      <div className={styles.jobcardtitle}>
        {props.post ? props.post.title : null}
      </div>
      <div className={styles.jobcardsubtitle}>
        {props.post ? props.post.desc : null}
      </div>
      <div className={styles.jobdetailbuttons}>
        {props.post
          ? props.post.tags?.map((tag) => (
              <button
                className={`${styles.searchbuttons} ${styles.detailbutton}`}
              >
                {tag}
              </button>
            ))
          : null}
      </div>
      <div className={styles.jobcardbuttons}>
        <button
          data-testid="handleApply"
          className={`${styles.searchbuttons} ${styles.cardbuttons}`}
          onClick={handleApply}
        >
          {props.post ? (props.post.applied ? "Applied" : "Apply Now") : null}
        </button>
        <button
          className={`${styles.searchbuttons} ${styles.cardbuttonsmsg}`}
          onClick={onEmailClick}
        >
          Contact
        </button>
      </div>
    </div>
  );
}

export default JobCards;
