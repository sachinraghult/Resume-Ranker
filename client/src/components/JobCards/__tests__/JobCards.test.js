import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import JobCards from "../JobCards";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <JobCards />
      </BrowserRouter>
    )
  );
});

it("renders correctly", () => {
  const data = {
    _id: "62cacefc225d2dcfbe60e8af",
    user: "62c9ea1df448b0664028a3c2",
    title: "Senior Manager",
    desc: "Check out latest 2020 Wells Fargo job vacancies in India. Get details on salary, company and location. Apply quickly to various Wells Fargo jobs ",
    exp: "Bachelor's degree in Engineering. \\nGood spoken and written English skills to effectively communicate technical concepts. \\nStrong logical, analytical skills and a systematic problem solving approach. \\nGood understanding of Software Development Life Cycle (SDLC).\\nDetail oriented, results driven, accountable and ability to work with multiple priorities.\\nAbility to research and report on a variety of issues using problem solving skills.\\nAbility to interact with integrity and a high level of professionalism with all levels of team members and management.\\nAbility to make timely and independent judgment decisions while working in a fast paced and results-driven environment",
    skills: [
      {
        skill: "Java",
        value: 99,
        _id: "62cacefc225d2dcfbe60e8b0",
      },
    ],
    tags: ["Full Time"],
    deadline: "2022-07-31T00:00:00.000Z",
    email: "wellsfargo@gmail.com",
    applications: [
      "62cb1db6ab7429104fb50d62",
      "62cb1df2ab7429104fb50d99",
      "62cb1ef0ab7429104fb50e49",
      "62cb1f74ab7429104fb50ea2",
    ],
    createdAt: "2022-07-10T13:07:08.477Z",
    updatedAt: "2022-07-10T18:50:28.205Z",
    __v: 8,
    keywords: [],
    applied: false,
  };
  const { getByTestId } = render(
    <BrowserRouter>
      <JobCards post={data} />
    </BrowserRouter>
  );
  fireEvent.click(getByTestId("handleApply"));
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <JobCards />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
