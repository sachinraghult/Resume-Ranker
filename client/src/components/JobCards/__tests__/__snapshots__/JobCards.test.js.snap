// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="jobcard"
>
  <div
    className="jobcardheader"
  >
    <div>
      <img
        alt="job poster"
        className="cardImg"
        src="http://localhost/Hiring/one.jpg"
      />
    </div>
    <div
      className="deadline"
    >
      <div
        style={
          Object {
            "fontSize": "14px",
          }
        }
      >
        <b>
          Deadline
        </b>
      </div>
      <div>
        <time
          className="dateAsCalendar positionPixels"
        >
          <span
            className="weekday"
          />
          <span
            className="day"
          />
          <span
            className="month"
          />
          <span
            className="year"
          />
        </time>
      </div>
    </div>
  </div>
  <div
    className="jobdetailbuttons"
  />
  <div
    className="jobcardtitle"
  />
  <div
    className="jobcardsubtitle"
  />
  <div
    className="jobdetailbuttons"
  />
  <div
    className="jobcardbuttons"
  >
    <button
      className="searchbuttons cardbuttons"
      data-testid="handleApply"
      onClick={[Function]}
    />
    <button
      className="searchbuttons cardbuttonsmsg"
      onClick={[Function]}
    >
      Contact
    </button>
  </div>
</div>
`;
